experiments/filter_mutants.py

```diff
@@ -0,0 +1,160 @@
+                                 var, mid, actual, diff)
+                if not is_nearly_eq:
+                    logger.debug("difference for parameter [%s] exceeds threshold (+/-%f)",
+                                     var, all_vars[var].noise)
+                    useful_inconsistent_traces.append((truth_fn, candidate_fn, diff/tolerance))
+                    broken = True
+                    break
+            if broken:
+                break
+        if not broken:
+            useless_inconsistent_traces.append((truth_fn, candidate_fn))
+    if useful_inconsistent_traces:
+        useless_inconsistent_traces.extend(list(c.fn_consistent_traces))
+        max_diff = max(useful_inconsistent_traces, key=lambda k: k[2])[2]
+        logger.debug('max_diff %f', max_diff)
+        return DatabaseEntry(c.diff,
+                            tuple([(o, t) for o, t, _ in useful_inconsistent_traces]),
+                            tuple(useless_inconsistent_traces)), max_diff
+    return None, 0.0
+
+
+def setup_logging(verbose: bool = False) -> None:
+    log_to_stdout = logging.StreamHandler()
+    log_to_stdout.setLevel(logging.DEBUG if verbose else logging.INFO)
+    logger.addHandler(log_to_stdout)
+    logging.getLogger('houston').addHandler(log_to_stdout)
+
+
+def parse_args():
+    p = argparse.ArgumentParser(description=DESCRIPTION)
+    p.add_argument('database', type=str, help='path to a database yaml file.')
+    p.add_argument('output', type=str, help='path to output database yaml file.')
+    p.add_argument('--top-n', type=int, default=0,
+                   help='return top n most suspicious mutants.')
+    p.add_argument('--threads', type=int, default=1,
+                   help='number of threads')
+    p.add_argument('--tolerance-factor', type=float, default=1.0,
+                   help='tolerance factor')
+    p.add_argument('--verbose', action='store_true',
+                   help='increases logging verbosity.')
+    return p.parse_args()
+
+
+def main():
+    args = parse_args()
+    setup_logging(args.verbose)
+
+    assert args.top_n >= 0
+
+    with open(args.database, 'r') as f:
+        db = YAML().load(f)
+
+    candidate_mutants = [DatabaseEntry.from_dict(e) for e in db['entries'] if e['inconsistent']]
+    logger.info("starting with %d mutants", len(candidate_mutants))
+
+    filtered_mutants = []
+    futures = []
+    with concurrent.futures.ProcessPoolExecutor(args.threads) as e:
+        for c in candidate_mutants:
+            future = e.submit(filter_mutants, c, args.tolerance_factor)
+            futures.append(future)
+
+        logger.debug("submitted all candidates")
+        for future in concurrent.futures.as_completed(futures):
+            entry, max_diff = future.result()
+            if entry:
+                filtered_mutants.append((entry, max_diff))
+        logger.debug("finished executing evaluating all mutants")
+ 
+    sorted(filtered_mutants, key=lambda k: k[1], reverse=True)
```

Comment:
I'm not sure what's going on here. If I remember correctly, `sorted` produces a new (sorted) list. We don't seem to be saving the result of this computation?